File: 프로그래머스/lv3/42898. 등굣길/등굣길.py
def solution(m, n, puddles):
    board = [[0] * m for _ in range(n)]
    for x, y in puddles:
        board[y - 1][x - 1] = -1

    board[0][0] = 1
    for i in range(n):
        for j in range(m):
            if board[i][j] == -1:
                board[i][j] = 0
                continue
            if i != 0:
                board[i][j] += board[i - 1][j]
            if j != 0:
                board[i][j] += board[i][j - 1]
            board[i][j] %= 1000000007

    return board[n - 1][m - 1]

File: 프로그래머스/Lv.3/42898. 등굣길/등굣길.java
import java.util.Arrays;

class Solution {
    
    public int solution(int m, int n, int[][] puddles) {
        int[][] board = new int[n + 1][m + 1];
        board[1][1] = 1;
        
        for (int[] puddle : puddles) {
            board[puddle[1]][puddle[0]] = -1;
        }
        
        for (int x = 1; x <= n; x++) {
            for (int y = 1; y <= m; y++) {
                if (x == 1 && y == 1) {
                    continue;
                }
                if (board[x][y] == -1) {
                    board[x][y] = 0;
                    continue;
                }
                
                board[x][y] = (board[x - 1][y] + board[x][y - 1]) % 1_000_000_007;
            }
        }
        
        return board[n][m];
    }
}